src/integrationTest/java/io/aiven/kafka/connect/jdbc/oracle/VerifyDeleteIT.java

```diff
@@ -0,0 +1,317 @@
+                if (i < numberOfTombstoneRecords) {
+                    final ProducerRecord<String, GenericRecord> record = createTombstoneRecord(i, partition);
+                    sendFutures.add(producer.send(record));
+                }
+            }
+        }
+
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+    }
+
+    @BeforeEach
+    public void afterEach() throws SQLException {
+        executeSqlStatement(DROP_TABLE);
+        executeSqlStatement(CREATE_TABLE);
+    }
+
+    @Test
+    public void testDeleteTombstoneRecord() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+
+        // Start the sink connector
+        connectRunner.createConnector(sinkConnectorConfigForDelete());
+
+        sendTestData(3);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(18))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(3);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0")
+                            .value().isEqualTo("1")
+                            .value().isEqualTo("2");
+                });
+
+        // Send test data to Kafka topic (including a tombstone record)
+        sendTestDataWithTombstone(1);
+
+        assertNotEmptyPoll(Duration.ofSeconds(50));
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(18))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(2);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("1")
+                            .value().isEqualTo("2");
+                });
+    }
+
+    @Test
+    public void testWithJustTombstoneRecordInInsertMode() throws Exception {
+        // Test logic is similar to previous tests, but with tombstone records.
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+
+        // Start the sink connector
+        final Map<String, String> config = sinkConnectorConfigForDelete();
+        connectRunner.createConnector(config);
+
+        sendTestDataWithTombstone(2);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(18))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(0);
+                });
+    }
+
+    @Test
+    public void testMultiInsertMode() throws Exception {
```

Comment:
We don't actually send any tombstone records in this test. Maybe it would make more sense in the `VerifyInsertIT` suite, similar to the `testSinkConnectorInDeleteMode` case that's already in that suite?